OpenAPI / Swagger does not like ContextParam annotation in controller method

When I try to build my project with Jooby 2.8.10 / OpenAPI and I have a controller method like the following, `mvn clean package` fails. I did not try it without swagger - so I'm not sure if it is a general OpenApi issue. Is this a bug in the code or simply a restriction that should be documented?

```java
    @GET
    @Path("me")
    public User getUserInformation(@ContextParam User user) {
        return userService.getUserInformation(user);
    }
```
To get the build working, I need to change the method as follows:
```java
    @GET
    @Path("me")
    public User getUserInformation(Context ctx) {
        User user = ctx.attribute("user")
        return userService.getUserInformation(user);
    }
```
The output from the maven command is as follows:
``` 
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for br.com.goclip:reciprocidade:jar:1.0.0
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: io.jooby:jooby-swagger-ui:jar -> duplicate declaration of version 2.8.10 @ line 63, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] --------------------< br.com.goclip:reciprocidade >---------------------
[INFO] Building reciprocidade 1.0.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ reciprocidade ---
[INFO] Deleting /home/m/projects/reciprocidade-api/target
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-versions) @ reciprocidade ---
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ reciprocidade ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 4 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ reciprocidade ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 26 source files to /home/m/projects/reciprocidade-api/target/classes
[INFO] 
[INFO] --- jooby-maven-plugin:2.8.10:openapi (default) @ reciprocidade ---
[INFO] Generating OpenAPI: br.com.goclip.App
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.150 s
[INFO] Finished at: 2020-08-20T12:14:03-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.jooby:jooby-maven-plugin:2.8.10:openapi (default) on project reciprocidade: execution of openapi resulted in exception: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.jooby:jooby-maven-plugin:2.8.10:openapi (default) on project reciprocidade: execution of openapi resulted in exception
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoFailureException: execution of openapi resulted in exception
    at io.jooby.maven.BaseMojo.execute (BaseMojo.java:83)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: java.lang.NullPointerException
    at io.jooby.internal.openapi.RouteParser.checkParameters (RouteParser.java:166)
    at io.jooby.internal.openapi.RouteParser.parse (RouteParser.java:86)
    at io.jooby.openapi.OpenAPIGenerator.generate (OpenAPIGenerator.java:165)
    at io.jooby.maven.OpenAPIMojo.doExecute (OpenAPIMojo.java:60)
    at io.jooby.maven.BaseMojo.execute (BaseMojo.java:78)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

--- a/modules/jooby-openapi/src/main/java/io/jooby/internal/openapi/AnnotationParser.java
+++ b/modules/jooby-openapi/src/main/java/io/jooby/internal/openapi/AnnotationParser.java
@@ -175,6 +175,10 @@ public class AnnotationParser {
       "kotlin.coroutines.Continuation"
   ).stream().collect(Collectors.toSet());
 
+  static final Set<String> IGNORED_ANNOTATIONS = Arrays.asList(
+      ContextParam.class.getName()
+  ).stream().collect(Collectors.toSet());
+
   public static List<OperationExt> parse(ParserContext ctx, String prefix,
       Signature signature, MethodInsnNode node) {
     if (signature.matches(Class.class) ||
@@ -347,6 +351,13 @@ public class AnnotationParser {
         } else {
           annotations = Collections.emptyList();
         }
+
+        if (annotations != null && annotations.stream()
+            .anyMatch(n -> IGNORED_ANNOTATIONS.contains(ASMType.parse(n.desc)))) {
+
+          continue;
+        }
+
         ParamType paramType = ParamType.find(annotations);
 
         /** Required: */

--- a/modules/jooby-openapi/src/test/java/issues/i1934/App1934.java
+++ b/modules/jooby-openapi/src/test/java/issues/i1934/App1934.java
@@ -1,0 +1,9 @@
+package issues.i1934;
+
+import io.jooby.Jooby;
+
+public class App1934 extends Jooby {
+  {
+    mvc(new Controller1934());
+  }
+}

--- a/modules/jooby-openapi/src/test/java/issues/i1934/Controller1934.java
+++ b/modules/jooby-openapi/src/test/java/issues/i1934/Controller1934.java
@@ -1,0 +1,16 @@
+package issues.i1934;
+
+import examples.Person;
+import io.jooby.annotations.ContextParam;
+import io.jooby.annotations.GET;
+import io.jooby.annotations.Path;
+
+@Path("/openapi")
+public class Controller1934 {
+
+  @GET
+  @Path("me")
+  public Person getUserInformation(@ContextParam Person user) {
+    return user;
+  }
+}

--- a/modules/jooby-openapi/src/test/java/issues/i1934/Issue1934.java
+++ b/modules/jooby-openapi/src/test/java/issues/i1934/Issue1934.java
@@ -1,0 +1,22 @@
+package issues.i1934;
+
+import io.jooby.internal.openapi.ResponseExt;
+import io.jooby.openapi.OpenAPITest;
+import io.jooby.openapi.RouteIterator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class Issue1934 {
+
+  @OpenAPITest(value = App1934.class, ignoreArguments = true)
+  public void shouldParseContextParamArg(RouteIterator iterator) {
+    iterator.next(route -> {
+          assertNull(route.getParameters());
+          ResponseExt response = route.getDefaultResponse();
+          assertEquals("200", response.getCode());
+          assertEquals("Success", response.getDescription());
+          assertEquals("examples.Person", response.getJavaType());
+        }).verify();
+  }
+}

